sdk/servicebus/Azure.Messaging.ServiceBus/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,305 @@
+
+### Processor stops immediately
+
+This typically comes up for demo or testing scenarios. Calling StartProcessingAsync returns immediately after the processor has started. Calling this method will not keep your app alive for as long as the processor is running, so you'll need some other mechanism to keep the app alive until you wish to shut it down. For demos or testing, it may be sufficient to just add a Console.ReadKey() call after you start the processor. For production scenarios, you will likely want to use some sort of framework integration like [BackgroundService][BackgroundService] to provide convenient application lifecycle hooks that can be used for starting and disposing the processor.
+
+## Troubleshoot transactions
+
+For general information about transactions in Service Bus, see the [Overview of Service Bus transaction processing][Transactions].
+
+### Supported operations
+
+Not all operations are supported when using transactions. To see the list of supported transactions, see [Operations within a transaction scope][TransactionOperations].
+
+### Timeout
+
+A transaction times out after 2 minutes. The transaction timer starts when the first operation in the transaction starts.
+
+### Operations in a transaction are not retried
+
+This is by design. Consider the following scenario - you are attempting to complete a message within a transaction, but there is some transient error that occurs, e.g. ServiceBusException with Reason of ServiceCommunicationProblem. Suppose the request does actually make it to the service. If the client were to retry, the service would see two complete requests. The first complete will not be finalized until the transaction is committed. The second complete is not able to even be evaluated before the first complete finishes. The transaction on the client is waiting for the complete to finish. This creates a deadlock where the service is waiting for the client to complete the transaction, but the client is waiting for the service to acknowledge the second complete operation. The transaction will eventually timeout after 2 minutes, but this is a bad user experience. For this reason, we do not retry operations within a transaction.
+
+### Transactions across entities are not working
+
+In order to perform transactions that involve multiple entities, you'll need to set the ServiceBusClientOptions.EnableCrossEntityTransactions property to `true`. For details, see the [Transactions across entities][CrossEntityTransactions] sample. 
+
+## Quotas
```

Comment:
Interesting choice to include this as a stand-alone section since it is referenced inline for the exception scenarios.   What's the thinking?
_(context: Not objecting to it, but I am trying to decide whether to steal the idea for Event Hubs...)_

Reply:
Was trying to make this a one stop shop for all FAQ/troubleshooting. Quota questions are pretty common.